app/controllers/EHealthVDCController.scala

```diff
@@ -1,144 +1,163 @@
-      "prothrombinTime.value as prothrombinTime, totalWhiteCellCount.value as totalWhiteCellCount from joined where socialId='%s'".format(socialId)
-      val patientBloodTestsDF = spark.sql(query)
-      patientBloodTestsDF.limit(10).show(false)
-      patientBloodTestsDF.printSchema
-
-      val rawJson = patientBloodTestsDF.limit(10).toJSON.collect().mkString
-      Future.successful(Ok(Json.toJson(rawJson)))
-    }
-  }
 
   @ApiOperation(nickname = "getBloodTestComponentAverage",
     value = "Get average of component over an age range",
     notes =  "This method returns the average value for a specific blood test component in a specific age range, to be used by researchers. Since data are for researchers, patients' identifiers and quasi-identifiers won't be returned, making the output of this method anonymized.",
-    response = classOf[models.Patient], responseContainer = "List", httpMethod = "GET")
+    response = classOf[models.PatientAvg], responseContainer = "List", httpMethod = "GET")
   @ApiResponses(Array(
     new ApiResponse(code = 404, message = "Component never measured")))
-  def getTestAverage(@ApiParam(value = "component", required = true,
+  def getBloodTestComponentAverage(@ApiParam(value = "component", required = true,
     allowMultiple = false) testType: String,
                      @ApiParam(value = "startAgeRange", required = true,
                        allowMultiple = false) startAgeRange: Int,
                      @ApiParam(value = "endAgeRange", required = true,
-                       allowMultiple = false) endAgeRange: Int): Action[AnyContent] = {
-    Action.async {
+                       allowMultiple = false) endAgeRange: Int) = Action.async {
+    implicit request =>
       val spark = initService.getSparkSessionInstance
-      readData(spark)
-      val todayDate =  java.time.LocalDate.now
-      val minBirthDate = todayDate.minusYears(endAgeRange)
-      val maxBirthDate = todayDate.minusYears(startAgeRange)
-      val query = "select AVG(%s.value) from joined where birthDate>%d AND birthDate<%d".format(minBirthDate, maxBirthDate)
-      val patientBloodTestsDF = spark.sql(query)
-      patientBloodTestsDF.limit(10).show(false)
-      patientBloodTestsDF.printSchema
-
-      val rawJson = patientBloodTestsDF.limit(10).toJSON.collect().mkString
-      Future.successful(Ok(Json.toJson(rawJson)))
-    }  
+      val queryObject = request.body
+      var origtestType:String = testType
+      var newTestType:String = null
+      var avgTestType:String = null
+      if (origtestType.equals("cholesterol")) {
+        newTestType = "cholesterol_total_value"
+        avgTestType = "avg(cholesterol_total_value)"
+      } else {
+        newTestType = "%s_value".format(origtestType).replaceAll("\\.", "_")
+        avgTestType = "avg("+newTestType+")"
+      }
+
+      val queryToEngine = "SELECT patientId, date, %s FROM blood_tests".format(newTestType)
+
+      if (config.has("policy.enforcement.play.url")) {
```

Comment:
Same as before. This logic won't be duplicated when it's inside the "sendRequestToEnforcmentEngine.